File: LearningAgent/src/main/java/pl/edu/agh/agenty/learning/utils/NetworkProperties.java
package pl.edu.agh.agenty.learning.utils;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

/**
 * Created by Kurtzz on 05.05.2017.
 */
@Component
@ConfigurationProperties("network")
public class NetworkProperties {
    private int batchSize;
    private int randomSeed;
    private int numEpochs;

    public int getBatchSize() {
        return batchSize;
    }

    public void setBatchSize(int batchSize) {
        this.batchSize = batchSize;
    }

    public int getRandomSeed() {
        return randomSeed;
    }

    public void setRandomSeed(int randomSeed) {
        this.randomSeed = randomSeed;
    }

    public int getNumEpochs() {
        return numEpochs;
    }

    public void setNumEpochs(int numEpochs) {
        this.numEpochs = numEpochs;
    }
}

File: LearningAgent/src/main/scala/pl/edu/agh/agenty/learning/utils/NetworkProperties.scala
package pl.edu.agh.agenty.learning.utils

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

/**
  * Created by P on 08.06.2017.
  */
@Component
@ConfigurationProperties("network")
class NetworkProperties {
  var batchSize: Int = _
  var randomSeed: Int = _
  var numEpochs: Int = _

  def setBatchSize(value: Int): Unit = {
    batchSize = value
  }

  def setRandomSeed(value: Int): Unit = {
    randomSeed = value
  }

  def setNumEpochs(value: Int): Unit = {
    numEpochs = value
  }
}